src/test/java/temporary/tests/BaseTest.java

```diff
@@ -24,4 +23,46 @@ public void baseSetUp() {
     public static void baseTearDown() {
         quitWebDriver();
     }
+
+    @Test
+    public void checkFacebookRedirect(){
+        // Given
+        openPage();
+        // When
+        clickOnFacebookButton();
+        // Then
+        var driver = getWebDriverInstance();
+        var tabs = driver.getWindowHandles();
+        Assert.assertEquals("New tab is not opened", 2, tabs.size());
+        validateNewTabURL("https://facebook.com/");
+    }
+
+    @Test
+    public void checkTwitterRedirect(){
+        // Given
+        openPage();
+        // When
+        clickOnTwitterButton();
+        // Then
+        var driver = getWebDriverInstance();
+        var tabs = driver.getWindowHandles();
+        Assert.assertEquals("New tab is not opened", 2, tabs.size());
+
+        validateNewTabURL("https://twitter.com/");
+    }
+
+    @Test
+    public void checkLinkedinRedirect(){
+        // Given
+        openPage();
+        // When
+        clickOnLinkedinButton();
+        // Then
+        var driver = getWebDriverInstance();
+        var tabs = driver.getWindowHandles();
```

Comment:
u can change this like that: 
```java
     var tabs = getWebDriverInstance().getWindowHandles();
```

Reply:
@fnfldx @Fishgame1 What would the correct approach be here? The one I described below?

implement the interacting with different tabs in `WebDriverFactory`,
no need to use `getWindowHandles`, which means no need to invocate `getWebDriverInstance`  (as this will be supported in `WebDriverFactory`),
no need to assert, that the opened tabs' size increased > just `validateURL` on the current tab after switch